src/bokeh/core/property/dataspec.py

```diff
@@ -268,18 +274,24 @@ class NumberSpec(DataSpec):
 
     """
 
-    def __init__(self, default=Undefined, *, help: str | None = None, accept_datetime=True, accept_timedelta=True) -> None:
+    def __init__(self, default=Undefined, *, help: str | None = None, accept_datetime: bool = True, accept_timedelta: bool = True) -> None:
         super().__init__(Float, default=default, help=help)
+
         if accept_timedelta:
             self.accepts(TimeDelta, convert_timedelta_type)
+        else:
+            deprecated((3, 7, 0), "NumberSpec(..., accept_datetime=False)", "FloatSpec()")
+
         if accept_datetime:
             self.accepts(Datetime, convert_datetime_type)
+        else:
+            deprecated((3, 7, 0), "NumberSpec(..., accept_timedelta=False)", "FloatSpec()")
```

Comment:
What if someone want's to accept datetimes but not timedeltas, or vice versa? Maybe it's fine to just not support this scenario, and collapse the two cases going forward, but we should be explicit about that.

Reply:
There was no such case in all our usage of `NumberSpec`, so I doubt anyone else would need that. However, one can always use `FloatSpec` with and `.accepts()` clause for that one-off case. Ultimately I want to remove both `accept_*` arguments, perhaps in bokeh 4.0.